python/pysrc/kaskada/windows.py

```diff
@@ -31,33 +38,102 @@ class Since(Window):
 
     @staticmethod
     def minutely() -> Since:
-        """Return a window since the start of each minute."""
+        """Return a window since the start of each calendar minute."""
         return Since(predicate=lambda domain: Timestream._call("minutely", domain))
 
     @staticmethod
     def hourly() -> Since:
-        """Return a window since the start of each hour."""
+        """Return a window since the start of each calendar hour."""
         return Since(predicate=lambda domain: Timestream._call("hourly", domain))
 
     @staticmethod
     def daily() -> Since:
-        """Return a window since the start of each day."""
+        """Return a window since the start of each calendar day."""
         return Since(predicate=lambda domain: Timestream._call("daily", domain))
 
     @staticmethod
     def monthly() -> Since:
-        """Return a window since the start of each month."""
+        """Return a window since the start of each calendar month."""
         return Since(predicate=lambda domain: Timestream._call("monthly", domain))
 
     @staticmethod
     def yearly() -> Since:
-        """Return a window since the start of each year."""
+        """Return a window since the start of each calendar year."""
         return Since(predicate=lambda domain: Timestream._call("yearly", domain))
 
 
+@dataclass(frozen=True)
+class Tumbling(Window):
+    """Window producing a single value at each predicate.
+
+    "Tumbling" windows are a series of non-overlapping windows that produce values
+    each time the predicate evaluates to true. Each window is exclusive of the time
+    it starts at and inclusive of the end time.
+
+    Note:
+        Like other systems, Kaskada treats tumbling windows as non-overlapping.
+        When one window ends the next starts.
+
+        Unlike other systems, tumbling windows do not need to be of a fixed size.
+        Instead, they may be determined by a fixed duration, a calendar duration (such as a "month"),
+        or even a predicate.
+
+    Args:
+        predicate: the condition used to determine when the window resets.
+
+    Returns:
+        Window for aggregating cumulative values since the predicate.
```

Comment:
Tumbling windows don't return cumulative values. This should be `Window for aggregating values since the predicate`